File: roomdemo/src/main/java/com/example/roomdemo/MyApp.java
package com.example.roomdemo;

import android.app.Application;

public class MyApp extends Application {
    private static MyApp myApp;

    @Override
    public void onCreate() {
        super.onCreate();
        myApp = this;
    }

    public static MyApp getMyApp(){
        if (myApp == null){
            myApp = new MyApp();
        }
        return myApp;
    }
}

File: jetpack/src/main/java/com/example/jetpack/hilt/MyApp.kt
package com.example.jetpack.hilt

import android.app.Application
import dagger.hilt.android.HiltAndroidApp

@HiltAndroidApp
class MyApp :Application() {

    override fun onCreate() {
        super.onCreate()
        mApplication = this
    }

    companion object{
        lateinit var mApplication: MyApp
    }

}